File: hw/cevallos/u3/hw35StrategyPersistence/Java/strategyCRUD/src/main/java/ec/edu/espe/strategycrud/model/StorageStrategy.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Interface.java to edit this template
 */
package ec.edu.espe.strategycrud.model;

/**
 *
 * @author Mateo Cevallos
 */
public interface StorageStrategy {

    boolean addEvent(Event event);

    boolean updateEvent(Event event);

    boolean deleteEvent(String id);

    Event readEvent(String id);
}

File: hw/cevallos/u3/hw35StrategyPersistence/Python/calendar_app/strategy/storage_strategy.py
from abc import ABC, abstractmethod
from model.event import Event

class StorageStrategy(ABC):
    """Interfaz Strategy para operaciones de almacenamiento"""
    
    @abstractmethod
    def add_event(self, event: Event) -> bool:
        pass
    
    @abstractmethod
    def update_event(self, event: Event) -> bool:
        pass
    
    @abstractmethod
    def delete_event(self, event_id: str) -> bool:
        pass
    
    @abstractmethod
    def read_event(self, event_id: str) -> Event:
        pass